File: 1st year/FP/Lab/Test1/tester.py
from services import *

def test_add():
    flights = []

    try:
        add(flights, "", "", "", "")
        return False
    except:
        return True
    assert len(flights) == 0

    try:
        add(flights, "ab", "ab", "ab", "ab")
        return False
    except:
        return True
    assert len(flights) == 0

    try:
        add(flights, "abc", "ab", "ab", "ab")
        return False
    except:
        return True
    assert len(flights) == 0

    try:
        add(flights, "abc", "15", "ab", "ab")
        return False
    except:
        return True
    assert len(flights) == 0

    try:
        add(flights, "abc", "20", "ab", "ab")
        return False
    except:
        return True
    assert len(flights) == 0

    try:
        add(flights, "abc", "20", "abc", "ab")
        return False
    except:
        return True
    assert len(flights) == 0

    add(flights, "abc", "20", "abc", "abc")
    assert len(flights) == 1

test_add()

File: 1st year/POO/Exams/Practice/Tester.cpp
#include "Tester.h"

void AddCoordinator_WhenStudentHasACoordinator_TheCoordinatorIsNotAdded()
{
	// Arrange
	Service service{ "testTeachers.txt", "testStudents.txt" };

	// Act
	auto result = service.addCoordinator("teacher2", { "def", "cab", "Email2", 2020, "Title2", "teacher1" });

	// Assert
	assert(!result);
}

void AddCoordinator_WhenStudentHasNoCoordinator_TheCoordinatorIsAdded()
{
	// Arrange
	Service service{ "testTeachers.txt", "testStudents.txt" };

	// Act
	auto result = service.addCoordinator("teacher2", { "jkl", "Some Name", "Email4", 2022, "No title" });

	// Assert
	assert(result);
}

void SearchStudents_WhenSearchDoesNotMach_TheVectorIsEmpty()
{
	// Arrange
	Service service{ "testTeachers.txt", "testStudents.txt" };

	// Act
	auto result = service.searchStudents("mno");

	// Assert
	assert(result.empty());
}

void SearchStudents_WhenSearchingByValidId_TheVectorHasAStudent()
{
	// Arrange
	Service service{ "testTeachers.txt", "testStudents.txt" };

	// Act
	auto result = service.searchStudents("def");

	// Assert
	assert(result.size() == 1);
}

void SearchStudents_WhenSearchingByValidName_TheVectorHasAStudent()
{
	// Arrange
	Service service{ "testTeachers.txt", "testStudents.txt" };

	// Act
	auto result = service.searchStudents("abc");

	// Assert
	assert(result.size() == 2);
}

void testAll()
{
	AddCoordinator_WhenStudentHasACoordinator_TheCoordinatorIsNotAdded();
	AddCoordinator_WhenStudentHasNoCoordinator_TheCoordinatorIsAdded();
	SearchStudents_WhenSearchDoesNotMach_TheVectorIsEmpty();
	SearchStudents_WhenSearchingByValidId_TheVectorHasAStudent();
	SearchStudents_WhenSearchingByValidName_TheVectorHasAStudent();
}
